core/quality/mysterium_morqa.go

```diff
@@ -18,45 +18,145 @@
 package quality
 
 import (
+	"bytes"
 	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
 	"time"
 
 	log "github.com/cihub/seelog"
-	"github.com/mysteriumnetwork/node/requests"
+	"github.com/golang/protobuf/proto"
+	"github.com/mysteriumnetwork/metrics"
 )
 
 const (
 	mysteriumMorqaLogPrefix = "[Mysterium.morqa] "
+	mysteriumMorqaAgentName = "goclient-v0.1"
 )
 
+// HttpClient sends actual HTTP requests
+type HttpClient interface {
+	Do(*http.Request) (*http.Response, error)
+}
+
 // MysteriumMORQA HTTP client for Mysterium QualityOracle - MORQA
 type MysteriumMORQA struct {
-	http                 requests.HTTPTransport
-	qualityOracleAddress string
+	http    HttpClient
```

Comment:
why did we need redefine generic requests.HTTPTransport with HTTPClient?

Reply:
I have little different contract with this API for error messages, no pot reuse it.

Also I remade it with: :
- less code, means much simpler
- less dependencies, this client can be separate lib
- transport is mocked in tests